ui/lib/kv/addon/components/kv-patch-editor.hbs

```diff
@@ -0,0 +1,165 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<form {{on "submit" this.submit}} ...attributes>
+  <div class="flex column-gap-16 has-top-padding-s">
+    <Hds::Form::Label @controlId="newKey" class="one-fourth-width">
+      Key
+    </Hds::Form::Label>
+    <Hds::Form::Label @controlId="newValue" class="three-fourths-width">
+      Value
+    </Hds::Form::Label>
+  </div>
+
+  {{! Rows for existing keys (includes new rows after user clicks "Add") }}
+  {{#each this.patchData as |KV idx|}}
+    {{#let KV.key KV.value KV.state as |key value state|}}
+      <div class="flex column-gap-16 has-top-padding-s">
```

Comment:
Feels like we could break out the existing rows into a component so that it's easy to unit test outside of the form component?

Reply:
So each row would be something like this?
```hbs
<KvPatchRow 
 @updateKey={{this.updateKey}}
 @undo={{this.undo}}
 @originalSubkey={{this.originalSubkey}} 
/>
```